### PR TITLE
Residual dropout support

### PR DESCRIPTION
**Pull request by dvruette**
Adds functions for patching residual dropout into `GPTNeoX` models. Patching it like this should be fine since it won't be needed for inference, so we don't have to fork/subclass the model class. It should also be quite easy to add this to other models (Galactica, ...) if needed.

Residual dropout has been used during InstructGPT [1] training, presumably to help prevent overfitting:
> We train our SFT models for 16 epochs with residual dropout of 0.2.

I also include some config changes from the latest run

[1] https://arxiv.org/pdf/2203.02155.pdf

### CODE REVIEW

**Review by andreaskoepf — APPROVED** [review]
Thanks a lot! Let's give it a try ;-) ..
